pylinalg/func/misc.py

```diff
@@ -107,3 +107,48 @@ def aabb_transform(aabb, matrix, /, *, out=None, dtype=None):
     out[..., 1, :] = np.max(corners[..., :-1], axis=-2)
 
     return out
+
+
+def axis_angle_from_quaternion(quaternion, /, *, out=None, dtype=None):
+    """Convert a quaternion to axis-angle representation.
+
+    Parameters
+    ----------
+    quaternion : ndarray, [4]
+        A quaternion describing the rotation.
+    out : Tuple[ndarray, ...], optional
+        A location into which the result is stored. If provided, it
+        must have a shape that the inputs broadcast to. If not provided or
+        None, a freshly-allocated array is returned. A tuple must have
+        length equal to the number of outputs.
+    dtype : data-type, optional
+        Overrides the data type of the result.
+
+    Returns
+    -------
+    axis : ndarray, [3]
+        The axis around which the quaternion rotates in euclidean coordinates.
+    angle : ndarray, [1]
+        The angle (in rad) by which the quaternion rotates.
+
+    Notes
+    -----
+    To use `out` with a single quaternion you need to provide a ndarray of shape
+    ``(1,)`` for angle.
+
+    """
+
+    quaternion = np.asarray(quaternion)
+
+    if out is None:
+        quaternion = quaternion.astype(dtype)
+        out = (
+            quaternion[..., :3] / np.sqrt(1 - quaternion[..., 3] ** 2),
+            2 * np.arccos(quaternion[..., 3]),
+        )
+    else:
+        # Note: the elements of out must allow mutation, i.e., you can
```

Comment:
Looks like this sentence was not finished.

Reply:
Thought I had deleted this sentence. It became a note in the docstring.